tests/test_shadow_fork.py

```
@@ -73,31 +73,53 @@ def test_create_fails_if_already_active(self, tmp_path):
     def test_create_fails_if_base_branch_missing(self, tmp_path):
-        """Test that create fails if base branch doesn't exist."""
+        """Test that create falls back gracefully if base branch doesn't exist."""
         fork = ShadowFork(repo_root=tmp_path)
 
-        with patch.object(fork, "_run_git") as mock_git:
-            mock_git.side_effect = [
-                (True, "main", ""),  # Get current branch
-                (True, "", ""),  # Check for uncommitted changes
-                (False, "", "fatal"),  # Base branch doesn't exist
-            ]
+        def mock_git_side_effect(cmd, **kwargs):
```

Ruff (0.15.5)

[warning] Missing return type annotation for private function `mock_git_side_effect`

(ANN202)